Feature/include admin control panel in quick links of assistant

hashpalk opened the pull request:
Fixes #4953 

Changes made:
- Removed static `excludeFromAssistant: true` flag from Admin Control route
- Added dynamic permission check in `searchQuickLinks` method to filter out Admin Control based on `permissions.accessAdminControl`
- The Admin Control quick link will now only appear for users who have the required admin permissions

This ensures that:
- Admin users can still see and access Admin Control through the quick links
- Regular users without admin permissions will not see the Admin Control option in quick links
- The functionality is consistent with the main navigation bar behavior

dtslvr commented on libs/common/src/lib/routes/routes.ts:
```diff
@@ -21,7 +21,6 @@ export const internalRoutes: Record<string, IRoute> = {
     title: $localize`Settings`
   },
   adminControl: {
-    excludeFromAssistant: true,
```
```
excludeFromAssistant: (user: User) => {
  return hasPermission(
    user?.permissions,
    permissions.accessAdminControl
  );
};
```

dtslvr reviewed (changes requested):
Thanks for your contribution, @hashpalk.

Your solution addresses the problem. However, I would prefer an approach where the logic is defined in a single place (`routes.ts`).

If it helps, you could refactor e.g. `api` also to a functional concept:

```
api: {
  excludeFromAssistant: (_user: User) => {
    return true;
  },
  path: 'api',
  routerLink: ['/api'],
  title: 'Ghostfolio API'
},
```

Could you please iterate based on my suggestion?

hashpalk commented:
Good day @dtslvr. I have implemented the Admin Control quick link permission check using a functional approach in routes.ts. I also fixed the permission logic to correctly show or hide the link based on the user's accessAdminControl permission. The solution keeps all route-related logic centralized, as requested.

dtslvr commented on libs/ui/src/lib/assistant/assistant.component.ts:
```diff
@@ -627,6 +628,10 @@ export class GfAssistantComponent implements OnChanges, OnDestroy, OnInit {
     const allRoutes = Object.values(internalRoutes)
       .filter(({ excludeFromAssistant }) => {
+        if (isFunction(excludeFromAssistant)) {
+          return excludeFromAssistant(this.user);
```
I had to invert this to make it work

dtslvr reviewed (approved):
Thanks for your contribution, @hashpalk 👍🏻

I have slightly adapted the code. Ready to merge now.
